app/menus/projects/menu.rb

```diff
@@ -117,6 +121,11 @@ def shared_filters
 
     def persisted_filters
       @persisted_filters = ::ProjectQuery.visible(current_user).order(:name)
+        .partition { |query| query.id.in?(favored_ids) }.flatten
```

Comment:
We are partitioning the list and then directly flattening it again? Is this to move all favored queries to the front? If that's the case, than that's a really clever solution, but in a few weeks we will have forgotten what it does and why it works.

What do you think about adding a scope in the `Acts::Favoritable` module: 

```ruby
scope :with_favored_by_user, ->(user) {
  favorite = ::Favorite.arel_table

  join = arel_table
          .join(favorite, Arel::Nodes::OuterJoin)
          .on(
            favorite[:favored_type].eq(base_class.name),
            favorite[:favored_id].eq(arel_table[:id]),
            favorite[:user_id].eq(user.id)
          ).join_sources

  select(arel_table[Arel.star], "(favorites.id IS NOT NULL) AS favored").joins(join)
}
```

This will add a virtual attribute to the `ProjectQuery` that is loaded with that scope and we can use that for sorting:

```ruby
ProjectQuery.with_favored_by_user(current_user).order("favored DESC", "name ASC")
```

This returns favored queries first (PG is a bit dumb that it sorts `TRUE` as last fields).

The scope has an obvious name and additionally we don't have to do an additional query for the favorited IDs.

Reply:
The good thing is that it is "hidden" away in the concerns code and we can test the implementation/query generation once and then when we use it can just rely on it

Reply:
The code is covered, but acts as favorable is tested only implicitly, so I'll note to create tests for it